tests/device_tests/test_msg_signmessage.py

```diff
@@ -206,3 +206,42 @@ def test_signmessage(client, coin_name, path, script_type, address, message, sig
     )
     assert sig.address == address
     assert sig.signature.hex() == signature
+
+
+MESSAGE_LENGTHS = (
+    pytest.param("This is a very long message. " * 64, id="normal_text"),
+    pytest.param("ThisIsAMessageWithoutSpaces" * 64, id="no_spaces"),
+    pytest.param("ThisIsAMessageWithLongWords " * 64, id="long_words"),
```

Comment:
Hmm can you perhaps add another two cases like this but with lots of utf8 continuations?

Reply:
that would be better in a unit test for core, no?

anyway i might need to add unit tests for the `Span` class anyway so I'll keep that in mind

Reply:
Right, I should improve my unit test instead.

Reply:
Feel free to cherry-pick ad05f8ade48eacef18fe238c02792b125c755aa8 which has some more test cases.

Reply:
cherry-picked as 9cf0385b